### PR TITLE
Update releasing documentation to reflect improved processes

### PR DESCRIPTION
**Pull request by andschwa**
This update was driven by the desire that the `VERSION` file in the
`master` branch reflects the upcoming release, and does not imply that
commits in the `master` branch are part of the prior release.

We also introduced release candidates.

### CODE REVIEW

File: docs/Releasing.md [hunk 1]
```diff
@@ -7,32 +7,47 @@ example:
...
 
 Release Branch
 --------------
 
-After the version bump PR is merged to `master`, the release manager creates a
-branch off `master` named `v0.4.x`.
+After this is merged, a release branch named `v0.5.x` is created from the last
+commit in `master` before the version changed to `v0.6.x`. The VERSION file in
+this commit should currently have `v0.5.x` in it, and the first commit to the
+release branch is to change it to `v0.5.0-rc1`.
```
Comment (EMumau) [inline docs/Releasing.md]: Currently, appending "-rc1" to the version number in the VERSION file breaks the build.  More changes will be required to support release candidate versioning.

Comment (andschwa) [reply docs/Releasing.md]: Can you post how it breaks so we can fix it?

**Comment by andschwa** [comment]
Maybe replaced by #1603 (we need to combine them).

**Comment by andschwa** [comment]
Closing for #1603.

**Pull request closed without merging**